oauth2-authorization-server/src/main/java/org/springframework/security/config/annotation/web/configurers/oauth2/server/authorization/OAuth2AuthorizationServerConfigurer.java

```diff
@@ -427,11 +427,17 @@ private void initEndpointMatchers(ProviderSettings providerSettings) {
 
 	private static void validateProviderSettings(ProviderSettings providerSettings) {
 		if (providerSettings.getIssuer() != null) {
+			URI issuerUri;
 			try {
-				new URI(providerSettings.getIssuer()).toURL();
+				issuerUri = new URI(providerSettings.getIssuer());
+				issuerUri.toURL();
 			} catch (Exception ex) {
 				throw new IllegalArgumentException("issuer must be a valid URL", ex);
 			}
+			// rfc8414 https://datatracker.ietf.org/doc/html/rfc8414#section-2
+			if (issuerUri.getQuery() !=null || issuerUri.getFragment() != null) {
+				throw new IllegalArgumentException("issuer has no query and fragment components");
```

Comment:
Update "issuer has no query and fragment components" to

"issuer cannot contain query or fragment component"

Reply:
The message has bean changed